### ISSUE TITLE
possibility to download full oses versions of the game "by hands"

### ISSUE THREAD

**Issue by DI555** [Unvanquished/updater#100]
would be very kind if will be the possibility to download full oses versions of the game "by hands"
many thanks in advance!

**Comment by illwieckz** [comment]
Can you explicit your need? If you are looking for a complete zip, we have an universal zip that can be downloaded from there: https://unvanquished.net/download/ (search for “Universal Zip”) or directly using [this link](https://unvanquished.net/download/zip), there is also a torrent for this (see the download page).

But this way of downloading and installing the game **isn't recommended** because modern operating systems can makes wrong assumptions when extracting zips and may even forbid the game to run, and some configuration would not be done.

As written on the download page about the universal zip:

> This is for advanced users. You won't be able to use the [server list](https://unvanquished.net/servers/) from this website to join games,
> neither join games from [the chat](https://unvanquished.net/chat/).
> Your operating system or unarchiver may not extract the game in a way it can work.
> The game may not run if you don't know what “application translocation” means.

We cannot support games installed using the universal zip by hand because many actual operating systems do not fully support this method of installation to begin with.

**Comment by DI555** [comment]
no, no, great thanks, it what's needed ;)! i'll do it fine!
just didn't find the full packages on the dl page ..

**Comment by ghost** [comment]
Could you explain why you were misled? This would help us improve the website to avoid such issues.

Yes, I reopen this because I think the root cause is not fixed.

**Comment by DI555** [comment]
i just like the full packages.. and started to search it firstly..

if i could suggest some good feature for site, i would ask about ftp server (ftp(s)://ftp.unvanquished.com) or/and emulated ftp (http(s)://ftp.unvanquished.com).., where could be placed all the packages of the game and all maps with map editors wirh other stuff a?d resources...this could be enforce accessibility of the game!!!
thank you in advance!

**Comment by DI555** [comment]
btw, is it strong restriction for universal zip users???

" Using them (universal zip) you won’t be able to use the server list from this website to join games, nor to join games from the chat. "

could i be free of this restrictions installing and updating game "by hands" ???

**Comment by illwieckz** [comment]
This is not a restriction. The files and binaries distributed in the universal zip are the exact same files and binaries installed using the installer.

But since the game is not installed when not using the installer, your web browser or your chat software have no way to know the game exists neither where the game is stored on your computer, and then cannot run the game. So you cannot use the server list from the website to join games, nor to join games from the chat, because no one installer told other programs the game is now installed.

**Comment by illwieckz** [comment]
@DI555 I assume @bmorel is asking you what prevented you to find the universal zip on the website.

- Did you see the “Download” tab on the site header?
- Did you confused it with the “Download” button on the main page assuming it would be the same?
- Is the text “other download options” under the “Download” button on the main page too tiny?
- Is the “Universal Zip” name not meaningful enough?
- Is there something else that may have prevented you to find the universal zip?

Because it's important for us to understand how and why someone can fail to download the game to make sure it does not happen again (as we want more players and want players to be satisfied 🙂️)

**Comment by ghost** [comment]
> i just like the full packages..

This is understandable, and here I could see an actual feature request: maybe we could have the updater in the universal archive, and a way for it to install files from there.
This way, it would be possible to easily install unvanquished on a system which have no network access, that the current mechanism does not easily allows.

> could i be free of this restrictions installing and updating game "by hands" ???

What you need to understand is that there is no traditional full-installer. There's either an archive you unpack in a place to run the game manually, or a tool which will download said archive and do the system setup stuff.
Using the archive works well, as far as we know, but we prefer players to use the Updater (which is capable of bootstrap, and hence can also be considered an installer, but requires internet datasource currently) for 2 reasons:

* it will configure your system
* you will have the latest files even if we're too lazy to build a release

> @DI555 I assume @bmorel is asking you what prevented you to find the universal zip on the website.

Yes. So that we can be able to improve the situation, so that we have more bandwidth for actual dev and less wasted in unneeded support. Support is not a problem, but if nobody needs it, it's all the better, don't you think?

> if i could suggest some good feature for site

It's actually possible, but if you want that, then you shuold be able to find the feature with the current website state. Otherwise, you would not be able to make any use of the files you will have found.
Basically, the installer itself grabs it's files from http, and AFAIK they're all in the same place.

Understand this, though, before searching: *WE DO NOT RECOMMEND DOING THIS AT ALL*, especially for end users. This is why I will not give you the URI, I'm sure you'll understand.

**Comment by DI555** [comment]
its all , an installer, quite good and usable
but keen people should have a chance to dl the full zip package, start an .exe or .bat and/or tune the server settings by hands, imo

if talkind about dl page, i would just make another blue button(like with installer) but with universal zip .. so it will be qickly visible

**Comment by illwieckz** [comment]
> keen people should have a chance to dl the full zip package

keen people already have a chance to dl the full zip package, it always has been.

> i would just make another blue button(like with installer) but with universal zip

The universal zip just doesn't run on some systems (for example, macOS has strong restrictions preventing it to run, at all) so while we make sure this zip is available for those who wants to tinker with, and because that's a reference archive, we must not make it an obvious download or people would be misleaded to try it without knowing how o make it works and would think the game is broken while it is not.

There are already too much tutorials full of bad or even faulty instructions on the web misleading players to badly install the game and break their setup, even saying things that contradict the readme in the zip itself… If we put this zip on first page like the installer, people would have right to expect it to work as well as the installer, but it does not, so it would be a deception and we should not deceive players.

In fact on the download page the universal zip previously had a big blue button but people who got bad experience from it requested to make it less obvious because this had gave them a false impression it should work while we knew it would not work out of the box, and then, they had right to not be happy.

The universal zip only works for lucky people (those being lucky to run on Linux, for example), and well, even for years it may have not been possible to click on the `daemon` icon in the file browser on Linux and get it run because of a [bug in a shared component](https://gitlab.freedesktop.org/xdg/shared-mime-info/-/issues/11) of all distros…

**Comment by ghost** [comment]
And what about a universal archive which would have an updater able to install from a local source? Maybe a self-extracting or something like that?

**Comment by DI555** [comment]
self extrating is os specific (
due to modern browsers drop ftp.., what about smth. 
like  `http://ftp.unvanquished.com` or `http://dl.unvanquished.com` with ftp emulater structure ?
to support the way updating "by the hands" , i mean , - to go there, see dates of a zip packages, downloading and updating manually..

**Comment by illwieckz** [comment]
> what about smth. like `http://dl.unvanquished.com`

We already have https://dl.unvanquished.net and it lists all of our releases there: https://dl.unvanquished.net/release/
It has been true since the beginning.

---

_Note:_ I edited your comment to make the urls non-clickable as they link to domain parking pages. Our domain name ends with `.net`, not with `.com`.

**Comment by ghost** [comment]
> self extrating is os specific

As are executable binaries.
So even if no self extract, I guess (for linux) a `chmod -x *; chmod +x updater` would make it obvious that updater is *the* way to install (it would do the chmod +x) stuff, and for windows it's possible to play with extension renaming. As macOS hell, I don't know and it's already borked anyway, as it should be on the most closed system for a  FOSS project.
This way, it would be rather difficult for people to mess up?

**Comment by illwieckz** [comment]
We workaround the `chmod +x` requirement on Linux by shipping the Linux Updater in a zip that carries the executable bit.

**Comment by ghost** [comment]
I know. What I meant is, maybe it's possible to do an archive where only the updater is executable, and to have the updater able to select data from local source?
This would in practice be an archive with an installer in it, and I believe it would fix this ticket.

**Comment by illwieckz** [comment]
It is already wanted for the updater to select data from local source to solve another problem which is the “store integration problem” (see #90) where a given store (like [lutris](https://lutris.net/)) can ask the user where he wants to download the game. The solution proposed to solve that other problem is to check the presence of a file the store can place purposedly next to the updater to enable the data loading from local source, that would also be reusable for an _updater-in-a-release-zip_ scenario.

I assume this would work for Linux and Windows out of the box.

As a specific side of the problem, the application translocation problem in macOS comes from the fact the `pkg/` folder is outside of the “application directory” where the application directory is the one for the engine. The macOS user doesn't run a binary by clicking on a binary but on a folder containing the binary, meaning the `pkg/` folder is outside of the folder containing the binary… If the translocation problem applies for the engine `.app` to load data from the `pkg/` folder that is outside of the `.app`, I doubt the updater `.app` can read a file next to it as proposed, I also don't know if it would be able to actually run the engine `.app` next to it.

I know macOS is the real pain here and it's super annoying to consider it may somewhat enforces the way we also deal with other operating system but on the other hand I'm not sure we really want to make variants for every OS file layout specificity.

At some point the only practical solution to have a zip release that works out of the box everywhere would be to do a zip release for every operating system. We would also be able to do a zip release for all OS but macOS and one for macOS, but then, we would do _specific_ things with users from all other OS interacting with the game in a way and those from this OS would interact in another way, meaning support would be specific, meaning one not running macOS cannot provide support for one running macOS because they would not share the same experience.

The actual updater provided a practical answer to that “every user of the software (here, the updater) shares the same experience” need.

**Comment by ghost** [comment]
I see.
Well, the raw archive is already marked as "should only be used by people who know their OS".
I'm fine at supporting options which are not supported by a single, minor OS in term of desktop market share, minor OS in term of contributors. If macOS developers want to annoy people by making things hard, there's no reason to make efforts to provide their users hard-if-not-impossible-features, but this is not a reason to not have that very same feature for sane OS users.
I think we should see to get that file thingy done, maybe not for 0.53, but certainly soon, because it would bring many benefits, as you pointed out yourself. Because I definitely see being on a store as a huge benefit, even more important than allowing to easily install from a local source with ease.

I'm not sure for the universal zip though. Depends on size increment per OS*architecture, and about how many users such a combination can have, I'd say. For example, it would be weird to include binaries for netBSD (I'm not even sure things would work there, but that's for the example).

Or maybe it's possible to have the updater for macOS in a specific folder so that their stupid layout restriction is obeyed? I really don't know anything about macOS, except that they make terribly selfish technology and marketing choices, and care less about portability and interoperability than I care about flies.

**Comment by illwieckz** [comment]
We can definitely implement the updater config file sidecar indeed. But not for 0.53. :-)

Then we would have to decide if we want to also ship the updater in universal zip. I'm more in favor of considering the engine as a component like everything else, so I would even just rename the updater `Unvanquished.exe` because it is the entry point.

**Comment by ghost** [comment]
> But not for 0.53. :-)

I updated my text a few times to stay polite toward macOS' devs... this information disappeared in those :)

**Comment by DI555** [comment]
could i set up game servers by hands from the unzipped build?
to write the link(s) of servers list into some file, may be or smth. ...
or needed link(s) is in there by default?

**Comment by ghost** [comment]
Please avoid reusing an issue to ask unrelated questions. You can find many information about running a server [here](https://wiki.unvanquished.net/wiki/Server/Running)

**Comment by illwieckz** [comment]
For future questions you can ask them on the forums: https://forums.unvanquished.net/

Yes you can set up server from the zip.

**Comment by necessarily-equal** [comment]
I think the questions asked here have been answered. If some of you think having the updater install from a local directory is a desired feature, worth spending time on, then we can open an issue on the updater's repo for that. (I can open the issue if ppl comment that they want it, but I won't work on that because I'd rather improve the game instead)
